### PR TITLE
feat: add support for easier-to-install, manual-only turret mounts

### PR DESCRIPTION
**Pull request by chaosvolt**
### Required

- [X] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [X] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [X] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [X] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

This makes it easier to make use of vehicle-mounted guns without access to the electronics a turret mount needs, via basically bringing back the old ability to have turrets that can only be fired manually.

## Describe the solution

C++ changes:
1. In turret.cpp and vehicle.h, defined function `vehicle::is_manual_turret`. This function checks if any parts on the same tile as the turret have the `MANUAL` flag installed on it, returning true if so.
2. Also in turret.cpp, set functions `vehicle::turrets_aim_and_fire_single`, `vehicle::find_all_ready_turrets`, `vehicle::turrets_set_targeting`, and `vehicle::turrets_set_mode` to all exclude `is_manual_turret` turrets, so that manual-only turrets can't be fired remotely using vehicle controls.
3. In vehicle.cpp, set `vehicle::can_mount` to forbid installing turret control units on manual-only turrets.
4. In vehicle_use.cpp, `vehicle::use_controls`'s section for turret UI now checks all turrets on the vehicle, and only shows options for remote turret behavior if at least one turret is available that's on a normal turret mount.

JSON changes:
1. Defined vehiclepart flag `MANUAL`, to be used by manual-only turret mounts, which don't require full electronic controls and welding to set up.
2. Defined vehicleparts for manual turret mounts. These are set to all use easier tool requirements than the welding called for by a turret mount, and use plain vehicle frames instead of a specialized item. Whether they're foldable or not also varies depending on whether the relevant frame is foldable.
3. Converted jeeps, technicals, 3-inch cannons, and gatling carriages to use manual turrets.
5. Misc: Removed another goddamn redundant JSON entry from No Hope in the process of checking mods for new turrets to support manual versions of.

## Describe alternatives you've considered

Shifting the ability to remote-fire a turret at all behind having a control unit on it.

## Testing

1. Checked affected JSON changes for syntax and lint errors.
2. Compiled and load-tested.
3. Spawned in a car and installed a simple turret mount on it.
4. Confirmed I can install a M249 on it, that I can load and unload the belt from it, and fire it manually.
5. Confirmed that vehicle controls don't allow me to fire it remotely.
7. Installed another M249, this time on a regular turret mount, confirmed I can still load, unload, and manually fire it just fine.
8. Tested that I'm allowed to install turret control unit on the M249 that's on a normal turret mount, but not on the manual one.
9. Confirmed remote firing options only let me mess with the one that's on a proper turret mount.
10. Spawned in a 3-inch cannon vehicle after, confirmed no odd behavior and can still load, unload, and fire it just fine.
11. Spawned in a technical and a humvee, confirmed nothing breaks and both can spawn loaded.
12. Checked affected C++ changes for astyle.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Aight. Per Scarf's feedback, instead of using dedicated manual-only copies of each turret we want to manually specify is manual-only, the focus is instead on whether the turret mount itself has the `MANUAL` flag. Affected turret mounts install the exact same turrets VP-wise as normal turret mounts, but they disallow remote firing and don't permit installing a turret control unit.

File: src/vehicle.h [hunk 1]
```diff
@@ -1331,6 +1331,10 @@ class vehicle
         turret_data turret_query( const tripoint &pos );
         turret_data turret_query( const tripoint &pos ) const;
 
+        /** Returns true if any part on the tile the turret is installed on has the MANUAL flag. */
+        bool is_manual_turret( vehicle_part &pt );
```
Comment (scarf005) [inline src/vehicle.h:1335]:
```suggestion
        bool is_manual_turret( vehicle_part &pt ) const;
```

shouldn't this be marked as `const` too?

File: src/turret.cpp [hunk 2]
```diff
@@ -78,6 +78,21 @@ turret_data vehicle::turret_query( const tripoint &pos ) const
...
 
+bool vehicle::is_manual_turret( vehicle_part &pt )
+{
+    for( const int elem : parts_at_relative( pt.mount, false ) ) {
+        if( part_info( elem ).has_flag( "MANUAL" ) ) {
+            return true;
+        }
+    }
+    return false;
+}
```
Comment (scarf005) [inline src/turret.cpp:89]:
```suggestion
bool vehicle::is_manual_turret(const vehicle_part &pt)
{
    const auto &parts = parts_at_relative(pt.mount, false);
    return std::any_of(parts.begin(), parts.end(), 
        [this](int elem) { return part_info(elem).has_flag("MANUAL"); });
}
```
let's use [std::any_of](https://en.cppreference.com/w/cpp/algorithm/all_any_none_of)

File: src/vehicle.h (hunk 1 above)
Comment (scarf005) [reply src/vehicle.h]: note that relevant code changes should be made too.

Comment (chaosvolt) [reply src/vehicle.h]: Oh shit, hecc.

File: src/turret.cpp [hunk 3]
```diff
@@ -78,6 +78,20 @@ turret_data vehicle::turret_query( const tripoint &pos ) const
...
+    return std::any_of( parts.begin(), parts.end(),
+    [this]( int elem ) {
+        return part_info( elem ).has_flag( "MANUAL" );
+    } );
+}
+
+bool vehicle::is_manual_turret( const vehicle_part &pt ) const
+{
+    return const_cast<vehicle *>( this )->is_manual_turret( const_cast<vehicle_part &>( pt ) );
+}
```
Comment (scarf005) [inline src/turret.cpp:93]:
```suggestion
bool vehicle::is_manual_turret( const vehicle_part &pt ) const
{
    const auto &parts = parts_at_relative( pt.mount, false );
    return std::any_of( parts.begin(), parts.end(),
    [this]( int elem ) {
        return part_info( elem ).has_flag( "MANUAL" );
    } );
}
```

do we actaully need non const param version here? wouldn't it be possible to pass in both const and non-const vehicle_part to const param?

Comment (chaosvolt) [reply src/turret.cpp:93]: Maybe, I wasn't sure. Will test in a sec.

Comment (chaosvolt) [reply src/turret.cpp:93]: Aight, compiles once the vehicle.h version of the definition is also deleted so.

**Review by scarf005 — APPROVED** [review]

**Pull request merged**